client/Components/Form.jsx

```diff
@@ -1,22 +1,86 @@
 import React from 'react';
-import GuestDropdown from './GuestDropdown.jsx';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { changeCheckin, changeCheckout } from '../store';
+import GuestDropdown from './GuestDropdown';
 import styles from './Form.css';
 
-const Form = (props) => {
-  const { maxGuests, handleSubmit } = props;
+const mapDispatchToProps = {
+  changeCheckin,
+  changeCheckout,
+};
+
+const mapStateToProps = (state) => {
+  const { checkin, checkout } = state.dateReducer;
+  let duration = 0;
+  if (checkin && checkout) {
+    const checkinDate = checkin.slice(-2);
+    const checkoutDate = checkout.slice(-2);
+    duration = Number.parseInt(checkoutDate, 10) - Number.parseInt(checkinDate, 10);
+  }
+  return { duration };
+};
+
+const ConnectedForm = (props) => {
+  const onCheckinChange = (e) => {
+    e.persist();
+    // const checkinDate = JSON.stringify(new Date(e.target.valueAsDate));
+
+    const checkin = e.target.value;
+    props.changeCheckin(JSON.stringify(checkin));
+    // Send action to Redux store to update checkin date
+
+    // console.log(e.target.value);
+  };
+  const onCheckoutChange = (e) => {
+    e.persist();
+    // const checkoutDate = JSON.stringify(new Date(e.target.valueAsDate));
+
+    const checkout = e.target.value;
+    props.changeCheckout(checkout);
```

Comment:
It looks like you didn't stringify your checkout here, in the same way that you did your checkin. Perhaps this is related to the bug we saw after standup.

Reply:
I just discovered this. It only works if neither are stringified.

Good catch!